apis/route53resolver/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+// CustomResolverEndpointParameters are custom parameters for ResolverEndpoint
+type CustomResolverEndpointParameters struct {
+	// The ID of one or more security groups that you want to use to control access
+	// to this VPC. The security group that you specify must include one or more
+	// inbound rules (for inbound Resolver endpoints) or outbound rules (for outbound
+	// Resolver endpoints). Inbound and outbound rules must allow TCP and UDP access.
+	// For inbound access, open port 53. For outbound access, open the port that
+	// you're using for DNS queries on your network.
+	SecurityGroupIDs []string `json:"securityGroupIds,omitempty"`
+	// SecurityGroupIDRefs is a list of references to SecurityGroups used to set
+	// the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDRefs []xpv1.Reference `json:"securityGroupIdRefs,omitempty"`
+
+	// SecurityGroupIDsSelector selects references to SecurityGroupID used
+	// to set the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDSelector *xpv1.Selector `json:"securityGroupIdSelector,omitempty"`
+
+	IPAddresses []*IPAddressRequest `json:"ipAddresses"`
+}
+
+// CustomResolverRuleParameters are custom parameters for CustomResolverRule
+type CustomResolverRuleParameters struct {
+	// ResolverEndpointIDRef is the reference to the ResolverEndpoint used
+	// to set the ResolverEndpointID
+	// +optional
+	ResolverEndpointIDRef *xpv1.Reference `json:"resolverEndpointIdRefs,omitempty"`
+	// ResolverEndpointIDSelector selects references to ResolverEndpoint used
+	// to set the ResolverEndpointID
+	// +optional
+	ResolverEndpointIDSelector *xpv1.Selector `json:"resolverEndpointIdSelector,omitempty"`
+}
+
+// CustomResolverQueryLogConfigParameters are custom parameters for CustomResolverQueryLogConfig
+type CustomResolverQueryLogConfigParameters struct {
+}
+
+// IPAddressRequest is used by ResolverEndpoint
+type IPAddressRequest struct {
+	IP *string `json:"ip,omitempty"`
```

Comment:
Is this field actually optional? Also, would you mind adding a docstring comment?

Reply:
It's not optional.

Reply:
@AmnaIrfan could you remove the `omitempty` then? :)

Reply:
I am sorry I thought we were talking about `IPAddressRequest`. 
So yes `IP` is optional.  `Subnet` is required but then I have that as optional since we have subnet references and selectors for that.